api/core/tools/provider/builtin/vanna/tools/vanna.py

```diff
@@ -111,9 +111,8 @@ def _invoke(
         # with "visualize" set to True (default behavior) leads to remote code execution.
         # Affected versions: <= 0.5.5
         #########################################################################################
-        generate_chart = False
-        # generate_chart = tool_parameters.get("generate_chart", True)
-        res = vn.ask(prompt, False, True, generate_chart)
+        allow_llm_to_see_data = tool_parameters.get("allow_llm_to_see_data", False)
+        res = vn.ask(prompt, False, True, False, allow_llm_to_see_data)
```

Comment:
```suggestion
        res = vn.ask(prompt, print_results=False, auto_train=True, visualize=allow_llm_to_see_data)
```
Avoid using positional boolean parameters in this case, set parameter by name for better readibility.

Reply:
The `visualize` parameter controls whether to generate plotly code. Is this the exact objective you want to achieve?

Reply:
Firstly, visualize and allow_llm_to_see_data are different, and therefore they should not be set as visualize=allow_llm_to_see_data. Due to the #4930, the current visualize is always False, so we are removing the generate_chart. Additionally, we are adding the allow_llm_to_see_data parameter to control whether to allow LLM to see the data. @bowenliang123

Reply:
> therefore they should not be set as visualize=allow_llm_to_see_data.

If so, which correct parameter you are setting?

Reply:
```
    def ask(
        self,
        question: Union[str, None] = None,
        print_results: bool = True,
        auto_train: bool = True,
        visualize: bool = True,  # if False, will not generate plotly code
    ) -> Union[
```

I didn't see `allow_llm_to_see_data` parameter in `vn.ask` method signature of vanna 0.5.5 according to poetry lockfile.

Reply:
Avoid using positional boolean parameters anyway.

Reply:
vn.ask(prompt, False, True, False, allow_llm_to_see_data)
print_result = False
auto_train = True
visualize = False
allow_llm_to_see_data = allow_llm_to_see_data

Reply:
No. I dont have a clue for having allow_llm_to_see_data in this method. The code block above is copied from vanna 0.5.5. Show your evidence.

Reply:
Are you trying to use the newly added parameter >0.5.5 anyway?
So please bump the vanna package version as well. And make sure your're installing it correctly. Otherwise, the vanna version in peotry lockfile is the alignment for our discussion.

Reply:
> Avoid using positional boolean parameters anyway.

No problem. Make sure you are using named boolean parameters.

Reply:
ok

Reply:
@bowenliang123 done

Reply:
Thanks. But could you minimize the changes in lockfile? Revert the poetry.lock file first and run `dev/sync-poetry` script to update the poetry lockfile.

Reply:
@bowenliang123 done

Reply:
LGTM.

Reply:
LGTM.